src/main/java/chess/controller/ChessController.java

```diff
@@ -14,48 +18,77 @@
         final GameCommand gameCommand = retryOnException(
-                () -> GameCommand.createFirstGameCommand(inputView.readGameCommand())
-        );
+                () -> GameCommand.createFirstGameCommand(inputView.readGameCommand()));
         if (gameCommand.isEnd()) {
             return;
         }
-        final ChessBoard chessBoard = new ChessBoard(ChessBoardInitializer.create());
+        final ChessBoard chessBoard = getChessBoard(boardRepository);
+        updateBoardChange(boardRepository, chessBoard);
         outputView.printChessBoard(chessBoard);
-        retryOnException(() -> playChess(chessBoard));
+        printPoints(retryOnException(() -> playChess(boardRepository)));
     }
 
-    private void playChess(final ChessBoard chessBoard) {
-        while (true) {
+    private ChessBoard getChessBoard(final BoardRepository boardRepository) {
+        if (boardRepository.count() != 0) {
+            return getAllDataFrom(boardRepository);
+        }
+        return new ChessBoard(ChessBoardInitializer.create());
+    }
+
+    private ChessBoard getAllDataFrom(final BoardRepository boardRepository) {
+        return boardRepository.findAll()
+                .convert();
+    }
+
+    private ChessBoard playChess(final BoardRepository boardRepository) {
+        ChessBoard chessBoard = getAllDataFrom(boardRepository);
+        while (!chessBoard.checkChessEnd()) {
             final GameArguments gameArguments = inputView.readGameArguments();
             final GameCommand gameCommand = gameArguments.gameCommand();
             if (gameCommand.isEnd()) {
                 break;
             }
+            if (gameCommand.isStatus()) {
+                printPoints(chessBoard);
+                continue;
+            }
             final MoveArguments moveArguments = gameArguments.moveArguments();
-            move(chessBoard, moveArguments);
+            chessBoard = move(chessBoard, moveArguments);
+            outputView.printChessBoard(chessBoard);
+            updateBoardChange(boardRepository, chessBoard);
         }
+        return chessBoard;
     }
 
-    private void move(final ChessBoard chessBoard, final MoveArguments moveArguments) {
+    private ChessBoard move(ChessBoard chessBoard, final MoveArguments moveArguments) {
         final ChessPosition source = moveArguments.createSourcePosition();
         final ChessPosition target = moveArguments.createTargetPosition();
-        chessBoard.move(source, target);
-        outputView.printChessBoard(chessBoard);
+        return chessBoard.move(source, target);
     }
 
-    private void retryOnException(final Runnable retryOperation) {
-        boolean retry = true;
-        while (retry) {
-            retry = tryOperation(retryOperation);
+    private void printPoints(final ChessBoard chessBoard) {
+        final PointCalculator pointCalculator = chessBoard.getPointCalculator();
+        final Points points = pointCalculator.calculate();
```

Comment:
```suggestion
        final Points points = chessBoard.calculatePoints();
```
PointCalculator 로 점수 계산 로직을 분리하는건 좋지만, board 에서 이를 외부로 주는 것이 아니라 보드가 직접 사용해서 점수를 반환하는 방식으로 만들어보는 것이 더 좋을 것 같아요.

Reply:
Board가 너무 많은 책임을 가지고 있어보여 점수 계산 로직의 분리를 시도해보았는데요.
동일한 필드를 공유하고 있어 어색해보였는데 혹시 이 부분이 캡슐화를 위반한 걸까요?

Reply:
캡슐화 위반...이라고 할 수도 있죠. 사실 그런 생각까지 하지는 않고 ChessBoard 를 얼마나 편리하게 사용할 수 있는지를 생각했어요.
점수를 계산할 수 있는 환경이 잘 마련되어 있는 ChessBoard 에서 점수 계산을 다 해서 주면 편리한데,
위의 구조에서는 ChessBoard 에게 계산기를 달라고 요청한 뒤, 그 계산기로부터 점수를 달라고 하는 **없어도 될 절차가 생겼거든요**
앞으로 체스 점수를 계산하기 위해서는 이 절차를 따라야한다는 불편함이 생기는거죠.

사실 이는 꽤나 흔히 발생하는 일이라고 생각해요.
만드는 사람 입장에서 보면 '이 객체는 하는 일이 너무 많아. 책임을 분산시켜야해' 라는 생각이 듭니다.
그래서 다른 객체를 추가하여 로직을 분산시키고, 그 작업을 완료하기까지 필요한 절차를 복잡하게 만들어요.
그러고 `클래스 파일에 담겨있는 코드 수가 줄어들어서`, 아니면 `public 메서드 수가 줄어들어서` 뭔가 책임을 잘 분산시켰다는 느낌을 받습니다. 하지만 유지보수하거나, 이를 활용하여 기능을 추가하는 사람 입장에서는 복잡도가 오히려 올라가는 셈이죠.

책임을 분산하는건 PointCalculator 를 만들고, ChessBoard 에서 이를 사용하는 것만으로도 충분한 거 같아요.
그런데 Calculator 를 아예 없애주셨군요....!

Reply:
아하.. PointCalculator를 분리한 시도는 나쁘지 않았었군요.
제가 피케이의 리뷰 의도를 잘못 파악했었네요. 아래처럼 이해한 내용이 맞을까요?

```java
public class ChessBoard {

    public Points calculate() {
        final PointCalculator pointCalculator = new PointCalculator();
        return pointCalculator.calculate();
    }


    private PointCalculator getPointCalculator() {
            return new PointCalculator(Collections.unmodifiableMap(board));
    }
}
```

저는 ChessBoard 와 PointCalculator 가 동일한 필드를 들고 있어서 (실제로 그렇게 넘겨주기도 하구요) 잘못된 책임 분리라 생각하였어요.
이러한 의도로 말씀하신 줄 알고 삭제하게 되었답니다..!

Reply:
제가 리뷰를 불명확하게 해서 그렇습니다 😭 
네, 저는 위에 다온이 적어주신 방식을 의미한거긴 했어요.

하지만 명확하게 전달하고 싶은 말은, '만약 PointCalculator 로 분리를 한다면' 위와 같은 방식으로 ChessBoard 안에서 PointCalculator 를 사용하는 방식으로 만드는게 좋아보여요.

PointCalculator 로 분리하지 않는 방식도 저는 충분히 좋다고 생각해요.
메서드가 많지만 다온은 public 한 비즈니스 로직을 가장 위로 올려두고 있기도 하구요!

Reply:
> 하지만 유지보수하거나, 이를 활용하여 기능을 추가하는 사람 입장에서는 복잡도가 오히려 올라가는 셈이죠.

피케이의 의도는 점수를 계산하는 로직이 2가지로 분리되어 코드를 이해하는데 불필요한 절차가 생겼다는 말씀이시겠군요..!! 무슨 말씀이신지 명확히 이해했습니다!